hawk/src/main/java/com/orhanobut/hawk/SqliteStorage.java

```diff
@@ -154,32 +164,47 @@ public synchronized boolean contains(String key) {
 
     public synchronized String get(String key) {
       SQLiteDatabase db = this.getReadableDatabase();
-      Cursor cursor = db.rawQuery("SELECT * FROM " + TABLE_NAME +
-          " WHERE " + COL_KEY + " = '" + key + "'", null);
-      if (cursor == null) {
-        return null;
-      }
-      cursor.moveToFirst();
-      if (cursor.getCount() == 0) {
-        return null;
+      Cursor cursor = null;
+      String value = null;
+      try {
+        cursor = db.rawQuery("SELECT * FROM " + TABLE_NAME +
+                " WHERE " + COL_KEY + " = '" + key + "'", null);
+        if (cursor == null) {
+          return null;
+        }
+        if (!cursor.moveToFirst()) {
+          return null;
+        }
+        value = cursor.getString(1);
+      } finally {
+        if (cursor != null) {
+          cursor.close();
+        }
+        db.close();
       }
-      String value = cursor.getString(1);
-      cursor.close();
-      db.close();
       return value;
     }
 
     public synchronized boolean clearAll() {
       SQLiteDatabase db = this.getWritableDatabase();
-      db.execSQL("DELETE FROM " + TABLE_NAME);
-      db.close();
+      try {
+        db.execSQL("DELETE FROM " + TABLE_NAME);
+      } catch (SQLException ignored) {
+        return false;
```

Comment:
same situation as above here, in catch clause db won't be closed and wider exception, maybe even just Exception catch would be good